### ISSUE TITLE
Read Color QR code

### ISSUE THREAD

**Issue by duttabhishek0** [journeyapps/zxing-android-embedded#693]
#### Not able to read QR codes that are colored and have a logo in between

**Which library version are you using?**
v4.3.0

**Which phone/tablet are you using, and which Android version does it run?**
Tested on Samsung A51 and Samsung Note 20(API 31/Android 11)

**Does the same happen on other devices or an emulator?**
yes

**In the case of an error do you have a stack trace or adb logs?**
NA

I want to scan QR codes that are colored and have a logo in between. Unfortunately, nothing happens the scanner just scans unless and until I go back.

**Comment by rkistner** [comment]
1. Can you post a sample of the barcode that fails to scan?
2. Does it work in other barcode scanners, such as [this one](https://play.google.com/store/apps/details?id=com.google.zxing.client.android&hl=en_ZA&gl=US)?

In general, QR codes should have a dark foreground on a light background. The color shouldn't matter, as long as the contract is high enough.

If the QR code has a light foreground on dark background, you'll have compatibility issues with many barcode scanners. With this library, you can use the "INVERTED_SCAN" or "MIXED_SCAN" modes - see this:
https://github.com/alpbak/zxing-android-embedded/blob/e48229ac3f47318e4b5516c645147d7dea37dc28/sample/src/main/java/example/zxing/MainActivity.java#L42

If there is a logo in between, there is also a chance that it removes too much info to be able to scan properly.

**Comment by duttabhishek0** [comment]
1. Can you post a sample of the barcode that fails to scan?
![image](https://user-images.githubusercontent.com/56694152/151844088-9e6de1d1-0875-4d60-83fd-1573425826c3.png)

2. Does it work in other barcode scanners, such as this one?
No

**Comment by duttabhishek0** [comment]
Yes I can change the contrast. I will do the needful. Thanks for help :)

**Comment by khavnu** [comment]
> 1. Can you post a sample of the barcode that fails to scan?
>    ![image](https://user-images.githubusercontent.com/56694152/151844088-9e6de1d1-0875-4d60-83fd-1573425826c3.png)
> 2. Does it work in other barcode scanners, such as this one?
>    No

You can use GlobalHistogramBinarizer instead of HybridBinarizer(ex: in MixedDecoder.java). This might affect to performance
Hope this help!

**Comment by henry-leu** [comment]
Hi,

I also have the same issue reading the qr image.
My company has this greenish qr image.
The qr image contains JWT token string.
ZXing is able to read it. 
But the parse result is not a JWT string.
It is an arbitrary string.

May I know what the issue is?
Could it be the image size too big?
Color of the image?

Thanks!

**Comment by rkistner** [comment]
If it scans and gives a result, the issue is not likely with the color or size of the image.

Check what results you get with other scanners. It's possible that it's an issue with how the barcode was encoded, rather than with the scanning. If you do get different results with other scanners, create a new issue here with a sample of the barcode.

**Comment by henry-leu** [comment]
I use the built-in scanner on iPhone.
It gave me a correct JWT token text.

**Comment by emartynov** [comment]
So any recommendation what to do with QR code in the dark mode?

I tried the solution above with the MIXED scan type but it didn't work.

**Comment by emartynov** [comment]
Let me put an example here:
<img width="198" alt="Screenshot 2024-07-05 at 17 08 22" src="https://github.com/journeyapps/zxing-android-embedded/assets/295424/2751be51-1826-447c-98ff-bc0f7f637f73">

~This is recognised by Android camera app but not with this library.~

Actually, I made a mistake and MIXED mode is working for the QR above.
